uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -299,12 +318,33 @@ public static function sendDailyReminder() {
         'settings' => $settings,
       ]);
 
-      self::_send($contactsResult->contact_id, $contactsResult->email, 'Daily Reminder', $templateBodyHTML);
+      self::_send($contactData['contact_id'], $contactData['email'], 'Daily Reminder', $templateBodyHTML);
     }
 
     return true;
   }
 
+  /**
+   * Returns contacts with active contracts from the list of
+   * contacts passed in.  If the job contract extension is not
+   * enabled the contacts passed in are returned as is.
+   *
+   * @param array $contactID
+   *
+   * @return array
+   */
+  private static function getContactsWithActiveContracts(array $contactID) {
+    if (self::$_relatedExtensions['hrjobcontract']) {
+      $result = civicrm_api3('HRJobContract', 'getcontactswithcontractsinperiod',[
+        'contact_id' => ['IN' => $contactID],
+      ]);
+
+      return array_column($result['values'], 'id');
+    }
+
+    return $contactID;
+  }
+
```

Comment:
Please add tests to cover these changes

Reply:
After discussing with @tunbola, it turns out that writing tests to this functionally will be much much more complex than I originally expected. The reason is that it is a highly coupled code, which dozens of dependencies that make the test setup very complex (there is even a piece of code [making queries directly to the Drupal database](https://github.com/compucorp/civihr-tasks-assignments/blob/a658cc58bee4a56984d796d5171fbb752cb6a885/uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php#L426-L436)!). To be able to test this, a considerable refactoring will be necessary, but that only increases the risk of introducing a new bug (because we don't have tests for the existing functionality). Given all that, I decided to make an exception this and I'm approving without the addition of tests.